src/integrations/clearml/__init__.py

```diff
@@ -0,0 +1,120 @@
+    external_url_columns: Optional[List[str]] = None,
+    dataset_project: Optional[str] = None,
+    dataset_name: Optional[str] = None,
+    parent_dataset_ids: Optional[List[str]] = None,
+    verbose: bool = False
+) -> None:
+    """Registers a new Dataset to ClearML containing files annotated by Toloka
+    Args:
+        toloka_df: pandas Dataframe containing data from Toloka
+        toloka_dataset_path: path to store `toloka_df` locally. This is a temp file used to register to clearml
+        dataset_project: Project containing the dataset in ClearML.
+        dataset_name: Naming the new dataset in ClearML
+        parent_dataset_id: Parent dataset ID
+        external_url_columns: columns containing external urls. If specified, the URLs in the columns will be added to the clearml Dataset
+                              as links and can be retrieved when calling get_clearml_dataset
+        verbose: If True print to console files added/modified
+    Examples:
+        The example shows how register  a new Dataset in ClearML.
+        >>> answers_df = toloka_client.get_assignments_df(pool_id='1')
+        >>> register_to_clearml(toloka_df=answers_df,
+        >>>             dataset_project='ClearMLProject', dataset_name='TolokaDataset')
+        ...
+        The example shows how to register a new Dataset in ClearML as a child Dataset.
+        >>> answers_df = toloka_client.get_assignments_df(pool_id='1')
+        >>> parent_dataset_id = # id of Dataset from ClearML
+        >>> register_to_clearml(parent_dataset_ids=parent_dataset_id, toloka_df=answers_df,
+        >>>                     dataset_project='ClearMLProject', dataset_name='TolokaDataset')
+        ...
+    """
+    if not (all([dataset_project, dataset_name]) or parent_dataset_ids):
+        raise ValueError("Either dataset_project and dataset_name or parent_dataset_id should be provided")
+
+    if not parent_dataset_ids:
+        dataset = Dataset.create(
+            dataset_project=dataset_project,
+            dataset_name=dataset_name
+        )
+    else:
+        parent = Dataset.get(dataset_id=parent_dataset_ids)
+        dataset = Dataset.create(
+            dataset_project=parent.project if not None else parent.project,
+            dataset_name=parent.name if not None else parent.name,
+            parent_datasets=[parent_dataset_ids]
+        )
+
+    toloka_csv_file = toloka_dataset_path if toloka_dataset_path is not None else '{}/toloka_ds_temp.csv'.format(tempfile.gettempdir())
+    toloka_df.to_csv(toloka_csv_file, index=False)
+    dataset.add_files(path=toloka_csv_file, verbose=verbose)
+
+    if external_url_columns is not None:
+        for column_name in toloka_df.columns:
+            # if column_name.startswith('INPUT:') or column_name.startswith('GOLDEN:') or column_name.startswith('OUTPUT:'):
+            if column_name in external_url_columns:
+                sample_value = toloka_df[column_name].iloc[0]
```

Comment:
This check is unfriendly to datasets with optional values (e.g. some answers may contain url while some will not) as the first row may happen to be empty. I suggest to restructure the method: first, download assignments from Toloka, then validate if every row in external_url_columns is either URL or empty. Then continue from line 89 but this time all non-empty external column values should be added to the dataset unconditionally